glommio/src/sys/uring.rs

```diff
@@ -1667,7 +1667,10 @@ impl Reactor {
             // is a timer set, we need to make sure we wake up to handle it.
             if let Some(dur) = user_timer {
                 self.timeout_src.set(Some(lat_ring.arm_timer(dur)));
-                flush_rings!(lat_ring)?;
+                if flush_rings!(lat_ring)? != 1 {
```

Comment:
agree that it isn't safe to sleep upon failures.
How reproucible is this ? A test would be great, but it may be hard to craft one

Reply:
This is me being paranoiac. I can't reproduce this but reading the man page there's no guarantee the kernel will consume what we give it. Maybe today it always does, but who knows about tomorrow :shrug: 

I'm especially concerned about cases where some internal data structure in the kernel is full, and it decides to apply back-pressure by denying new submissions for a while.